### PR TITLE
Use Okina in Hawaiʻi and ā with macron in Mānoa

### PR DESCRIPTION
**Pull request by weiji14**
**Description of proposed changes**

Replacing apostrophe with Okina letter in a few places. Approximating the Okina letter with a ` (octal code 140) for the roads gallery example.

**Preview** at https://pygmt-dev--2644.org.readthedocs.build/en/2644/gallery/lines/roads.html

Reference:
- https://en.wikipedia.org/wiki/%CA%BBOkina

Fixes https://github.com/GenericMappingTools/pygmt/pull/1474#discussion_r1306286723

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: AUTHORS.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ The following people have contributed code and/or documentation to the project
 * [Leonardo Uieda](http://www.leouieda.com/) | [0000-0001-6123-9515](https://orcid.org/0000-0001-6123-9515) | University of Liverpool, United Kingdom
 * [Liam Toney](https://liam.earth/) | [0000-0003-0167-9433](https://orcid.org/0000-0003-0167-9433) | University of Alaska Fairbanks, USA
 * [Malte Ziebarth](https://github.com/mjziebarth) | [0000-0002-5190-4478](https://orcid.org/0000-0002-5190-4478) | GFZ German Research Centre for Geosciences, Germany
-* [Max Jones](https://github.com/maxrjones) | [0000-0003-0180-8928](https://orcid.org/0000-0003-0180-8928) | University of Hawai'i at Mānoa, USA
+* [Max Jones](https://github.com/maxrjones) | [0000-0003-0180-8928](https://orcid.org/0000-0003-0180-8928) | University of Hawaiʻi at Mānoa, USA
```
Comment (weiji14) [inline AUTHORS.md:21]: Hi @maxrjones, just checking if you're still happy to keep your affiliation with UoH?

File: examples/gallery/lines/roads.py [hunk 2]
```diff
@@ -24,10 +24,10 @@
 
 fig = pygmt.Figure()
 
-# Define target region around O'ahu (Hawai'i)
+# Define target region around Oʻahu (Hawaiʻi)
 region = [-158.3, -157.6, 21.2, 21.75]  # xmin, xmax, ymin, ymax
 
-title = r"Main roads of O\047ahu (Hawai\047i)"  # \047 is octal code for '
+title = r"Main roads of O\140ahu (Hawai\140i)"  # \140 is octal code for `, approximating the Okina ʻ
```
Comment (weiji14) [inline examples/gallery/lines/roads.py:30]: Technically \140 is a [backtick](https://en.wikipedia.org/wiki/Backtick) ` (U+0060), but this is the closest approximation I know to the Okina letter `ʻ` (U+02BB) as mentioned at https://github.com/GenericMappingTools/pygmt/pull/2641/files#r1306294445. Anyone who knows how to plot the Okina properly is welcome to chime in!

Comment (seisman) [inline examples/gallery/lines/roads.py:30]: If we're going to use backtick, then we don't need to use its octal code.
```suggestion
title = r"Main roads of O`ahu (Hawai`i)"  #  Approximating the Okina ʻ with `
```

**Review by michaelgrund — APPROVED** [review]

File: pygmt/clib/session.py [hunk 3]
```diff
@@ -1641,7 +1641,7 @@ def extract_region(self):
         0.00, 10.00, -20.00, -10.00
 
         Using ISO country codes for the regions (for example ``'US.HI'`` for
```
Comment (yvonnefroehlich) [inline pygmt/clib/session.py:1643]: It's only a minor detail, but for consistency, we also should use double quotation marks in comments.
Please also update line 1657.
```suggestion
        Using ISO country codes for the regions (for example ``"US.HI"`` for
```

**Review by yvonnefroehlich — APPROVED** [review]

File: pygmt/clib/session.py (hunk 3 above)
Comment (weiji14) [reply pygmt/clib/session.py]: Done at 8c57ea4148dc70cc59073329b0ca2a584b9af48c.

**Comment by weiji14** [comment]
Will leave this open for another day (until US Eastern time evening) before merge.

File: pygmt/clib/session.py (hunk 3 above)
Comment (yvonnefroehlich) [reply pygmt/clib/session.py]: Thanks!

**Pull request merged**